modules/imports/src/main/scala/org/dhallj/imports/ResolveImportsVisitor.scala

```diff
@@ -184,27 +185,31 @@ final private class ResolveImportsVisitor[F[_] <: AnyRef](
 
     def importNonLocation(imp: ImportContext, mode: ImportMode, hash: Array[Byte]) =
       for {
-        _ <- if (parents.nonEmpty) ReferentialSanityCheck(parents.head, imp) else F.unit
+        _ <- ReferentialSanityCheck(parents.head, imp)
         _ <- rejectCyclicImports(imp, parents)
         r <- resolve(imp, mode, hash)
       } yield r
 
     for {
-      imp <- if (parents.isEmpty) canonicalize(i) else canonicalize(parents.head, i)
+      imp <- canonicalize(parents.head, i)
       result <- if (mode == ImportMode.LOCATION) importLocation(imp) else importNonLocation(imp, mode, hash)
     } yield result
   }
 }
 
 private object ResolveImportsVisitor {
-  def apply[F[_] <: AnyRef: Sync: Client]: F[ResolveImportsVisitor[F]] =
-    Sync[F].map2(ImportCache[F]("dhall"), ImportCache[F]("dhallj"))(apply[F](_, _))
+  def apply[F[_] <: AnyRef: Sync: Client](relativeTo: Path): F[ResolveImportsVisitor[F]] =
```

Comment:
What do you think about keeping around the old method signatures as well, using `cwd` as above? MiMa doesn't like the missing `apply`s, even though this is a private object. Then we could remove them in 0.3.x.

Reply:
Oh wow. I didn't think it would care about a private object. Yeah, that's fine by me